src/vehicle_gui_base.h

```diff
@@ -35,19 +69,49 @@ struct BaseVehicleListWindow : public Window {
 	};
 
 	static const StringID vehicle_depot_name[];
-	static const StringID vehicle_sorter_names[];
-	static GUIVehicleList::SortFunction * const vehicle_sorter_funcs[];
+	static const StringID vehicle_group_by_names[];
+	static const StringID vehicle_group_none_sorter_names[];
+	static const StringID vehicle_group_shared_orders_sorter_names[];
+	static VehicleGroupSortFunction * const vehicle_group_none_sorter_funcs[];
+	static VehicleGroupSortFunction * const vehicle_group_shared_orders_sorter_funcs[];
 
 	BaseVehicleListWindow(WindowDesc *desc, WindowNumber wno) : Window(desc), vli(VehicleListIdentifier::UnPack(wno))
 	{
-		this->vehicles.SetSortFuncs(this->vehicle_sorter_funcs);
+		// TODO: retrieve the user preference from somewhere instead of intializing to GB_NONE for every new window.
```

Comment:
Could this be added? Probably need to ensure that grouped vehicle lists are still opened with GB_NONE.